internal/ethapi/transaction_args.go

```diff
@@ -276,16 +268,33 @@ func (args *TransactionArgs) toTransaction() *types.Transaction {
 		if args.AccessList != nil {
 			al = *args.AccessList
 		}
-		data = &types.DynamicFeeTx{
-			To:         args.To,
-			ChainID:    (*big.Int)(args.ChainID),
-			Nonce:      uint64(*args.Nonce),
-			Gas:        uint64(*args.Gas),
-			GasFeeCap:  (*big.Int)(args.MaxFeePerGas),
-			GasTipCap:  (*big.Int)(args.MaxPriorityFeePerGas),
-			Value:      (*big.Int)(args.Value),
-			Data:       args.data(),
-			AccessList: al,
+		if args.FeeCurrency != nil || args.GatewayFeeRecipient != nil {
```

Comment:
At minimum, if any celo specific field is present, it should use the celo dynamic fee tx.

Reply:
https://github.com/celo-org/celo-blockchain/blob/7979f2f91d88b07233c71d0d83ea23721d5c7d7c/internal/ethapi/transaction_args.go#L180
if `args.GatewayFeeRecipient != nil`, then `args.GatewayFee` must have been set

Reply:
Hmm. That's true locally, but I don't think that invariant elsewhere. This is something that works, but I'm not sure it's worth the effort to be clever vs just adding another check.